test/src/org/labkey/test/tests/targetedms/InstrumentSchedulingTest.java

```diff
@@ -0,0 +1,221 @@
+        ));
+        List<Map<String, Object>> projectPaymentMethods = projectPaymentMethodInsert.execute(createDefaultConnection(), getProjectName()).getRows();
+
+        InsertRowsCommand projectResearcherInsert = new InsertRowsCommand("targetedms", "projectResearcher");
+        projectResearcherInsert.setRows(Arrays.asList(
+                Map.of("Project", projects.get(0).get("Id"), "Researcher", schedulerUser1Id),
+                Map.of("Project", projects.get(1).get("Id"), "Researcher", schedulerUser1Id),
+                Map.of("Project", projects.get(1).get("Id"), "Researcher", schedulerUser2Id)
+        ));
+        List<Map<String, Object>> projectResearchers = projectResearcherInsert.execute(createDefaultConnection(), getProjectName()).getRows();
+
+        InsertRowsCommand instrumentRateInsert = new InsertRowsCommand("targetedms", "instrumentRate");
+        instrumentRateInsert.setRows(Arrays.asList(
+                Map.of("Instrument", instruments.get(0).get("Id"), "rateType", rateTypes.get(0).get("Id"), "fee", 100),
+                Map.of("Instrument", instruments.get(1).get("Id"), "rateType", rateTypes.get(1).get("Id"), "fee", 110)
+        ));
+        List<Map<String, Object>> instrumentRates = instrumentRateInsert.execute(createDefaultConnection(), getProjectName()).getRows();
+    }
+
+    @Test
+    public void testSchedule()
+    {
+        goToProjectHome();
+        clickAndWait(Locator.linkWithText("Your project list"));
+        assertTextPresent(PROJECT_1, PROJECT_2);
+        clickAndWait(Locator.linkWithText(PROJECT_1));
+        waitAndClickAndWait(Locator.linkWithText("Schedule instrument time"));
+
+        String yearMonth = Calendar.getInstance().get(Calendar.YEAR) + "-";
+        int month = (Calendar.getInstance().get(Calendar.MONTH) + 1);
+        if (month < 10)
+        {
+            yearMonth += yearMonth;
+        }
+        yearMonth += month;
+
+        scheduleInstrument(yearMonth + "-02");
+        scheduleInstrument(yearMonth + "-03");
+        scheduleInstrument(yearMonth + "-03", true);
+        scheduleInstrument(yearMonth + "-03");
+
+        assertProjectEventCounts(2, 0);
+
+        sleep(1000);  // Wait for the dialog to clear out of the way
+        doAndWaitForPageToLoad(() -> selectOptionByText(Locator.id("projectDropDown"), PROJECT_2));
+
+        scheduleInstrument(yearMonth + "-04");
+        assertProjectEventCounts(1, 2);
+
+        scheduleInstrument(yearMonth + "-05");
+        assertProjectEventCounts(2, 2);
+
+        sleep(1000);  // Wait for the dialog to clear out of the way
+        doAndWaitForPageToLoad(() -> selectOptionByText(Locator.id("instrumentDropDown"), INSTRUMENT_2));
+        scheduleInstrument(yearMonth + "-06");
+        assertProjectEventCounts(1, 0);
+
+        goToDashboard();
+        clickAndWait(Locator.linkWithText("All instrument calendar view"));
```

Comment:
In cases like this, you navigate and then immediately click on an element on the page, I've started using `waitAndClickAndWait`. It will wait a little bit for the element to show up before it tries to click it.